src/main/thrift/parquet.thrift

```diff
@@ -595,6 +607,43 @@ union ColumnOrder {
   1: TypeDefinedOrder TYPE_ORDER;
 }
 
+struct PageLocation {
+/** Offset of the page in the file **/
+  1: required i64 offset
+
+/** Size of the page, including header. The same as PageHeader.compressed_page_size **/
+  2: required i32 compressed_page_size
+
+/** Index within the RowGroup of the first row of the page **/
```

Comment:
This should be more clear about the requirements for using this index structure: Pages must be split on record boundaries to ensure that no records are split across pages. The repetition level of the first value in each page must be 0.

Reply:
Good point. Currently we don't require this to be the case for nested columns though.

Reply:
@lekv @rdblue have we achieve this for nested column? i mean split on record boundaries to ensure that no records are split across pages, thanks a lot!

Reply:
https://github.com/apache/parquet-mr/blob/330242b3cc2a746f3bbb939304becfa313ef1e53/parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriterBase.java#L123 i found this, thank you all